pkg/secrethub/credentials/key.go

```diff
@@ -0,0 +1,104 @@
+package credentials
+
+import (
+	"errors"
+
+	"github.com/secrethub/secrethub-go/internals/auth"
+	"github.com/secrethub/secrethub-go/internals/crypto"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/internals/http"
+)
+
+type Key struct {
+	key              *RSACredential
+	exportPassphrase Reader
+}
+
+func (k Key) Verifier() Verifier {
+	return k.key
+}
+
+func (k Key) Encrypter() Encrypter {
+	return k.key
+}
+
+func (k Key) Provide(httpClient *http.Client) (auth.Authenticator, Decrypter, error) {
+	return k.key, k.key, nil
+}
+
+func (k Key) Passphrase(passphraseReader Reader) Key {
```

Comment:
What does this do? And this type is missing godoc as well.

Reply:
Added a comnent